### PR TITLE
phone number sanitizer implemented

### PR DESCRIPTION
**Pull request by guneyin**

### CODE REVIEW

File: phone.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	return out
+}
+
+// SanitizePhoneNumber sanitizes given phone number
+func SanitizePhoneNumber(phone string) (string, error) {
+	ph := phone
+
+	ph = convertArabicToLatin(ph)
+
+	re1, err := regexp.Compile("[^0-9]")
```
Comment (ehsannm) [inline phone.go]: Do not use regexp.Compile everytime, it is a heavy function. 
maybe a private global variable which initialized in init() func.
```
var phoneRegEx *regexp.RegEx

func init() {
     phoneRegEx = regexp.MustCompile(`$[0-9]+`)     
}
```

File: phone.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	if err != nil {
+		return "", err
+	}
+
+	ph = re1.ReplaceAllString(ph, "")
+
+	re2, err := regexp.Compile("^0+")
```
Comment (ehsannm) [inline phone.go]: same as previous comment, Compile is expensive call we should not call it on every request

**Review by ehsannm — CHANGES_REQUESTED** [review]
I suggest you try to sanitize it in one round, currently you are using two rounds which is not efficient.

**Comment by guneyin** [comment]
I refactored the SanitizePhoneNumber func. Regexp.compile count reduced to one

**Review by ehsannm — APPROVED** [review]

**Pull request merged**